distributed/http/scheduler.py

```diff
@@ -41,6 +44,48 @@ def get(self, rest):
         self.write(responses3)  # TODO: capture more data of response
 
 
+class KeyStatus(RequestHandler):
+    """What is the given key up to?"""
+    def get(self, key):
+        if key in self.server.nbytes:
+            out = "ready"
+        elif key in self.server.waiting:
+            out = 'waiting'
+        elif any(key in a for a in self.server.processing):
+            out = 'processing'
+        elif any(key in a for a in self.server.stacks):
+            out = 'queued'
+        elif key in self.server.exceptions:
+            out = 'error'
+        else:
+            out = None
+        where = self.server.who_has.get(key, [])
+        where = ["%s:%s" % w for w in where]
+        self.write({'status': out, 'machines': where})
```

Comment:
tuples encode to lists - all we need ot know is whether to expect  list or a tuple (always the latter for keys)